app/src/main/java/com/devshish/internship/di/NetworkModules.kt

```diff
@@ -2,68 +2,63 @@ package com.devshish.internship.di
 
 import android.content.Context
 import com.devshish.internship.BuildConfig
-import com.devshish.internship.data.api.GeniusAuthApi
-import com.devshish.internship.data.api.GeniusProfileApi
-import com.devshish.internship.data.api.GeniusSearchApi
+import com.devshish.internship.data.api.genius.GeniusAuthApi
+import com.devshish.internship.data.api.genius.GeniusProfileApi
+import com.devshish.internship.data.api.genius.GeniusSearchApi
+import com.devshish.internship.data.api.lastfm.LastFmChartsApi
 import com.devshish.internship.data.repository.AuthUseCase
 import com.devshish.internship.data.repository.SearchAPIRepository
 import com.devshish.internship.data.repository.TokenRepositoryImpl
+import com.devshish.internship.di.Urls.*
 import com.devshish.internship.domain.usecase.IAuthUseCase
 import com.devshish.internship.domain.repository.ISearchSongsRepository
 import com.devshish.internship.domain.repository.ITokenRepository
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
 import okhttp3.Interceptor
 import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
 import org.koin.android.ext.koin.androidApplication
+import org.koin.core.qualifier.named
 import org.koin.dsl.module
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 
-val networkModule = module {
-
-    single<ITokenRepository> {
-        TokenRepositoryImpl(
-            sharedPref = get()
-        )
-    }
-
-    single<ISearchSongsRepository> {
-        SearchAPIRepository(
-            api = get()
-        )
-    }
-
-    single<IAuthUseCase> {
-        AuthUseCase(
-            api = get(),
-            tokenRepository = get()
-        )
-    }
+enum class Urls {
+    GENIUS,
+    LASTFM
+}
 
-    single {
-        provideAuthApiRepository(
-            retrofit = get()
-        )
-    }
+val networkModule = module {
```

Comment:
Maybe split this into 2 modules - like Genius module, and LastFm module? Because right now it is ind of mess 

And then add it to network module?